include/verilated.cpp

```diff
@@ -3098,3 +3098,18 @@ void VerilatedAssertOneThread::fatal_different() VL_MT_SAFE {
 #endif
 
 //===========================================================================
+// VlDeleter:: Methods
+
+void VlDeleter::deleteAll() {
+    VerilatedLockGuard lock{m_mutex};
+    while (!m_garbage.empty()) {
+        // Pop it in case a new object gets added by calling it
+        VlClass* const objp = m_garbage.back();
+        m_garbage.pop_back();
+        lock.unlock();
+        delete objp;
+        lock.lock();
+    }
```

Comment:
Is it faster to iterate on each object, then clear the container?

Reply:
Deleting or calling a destructor could decrement a member object's refcount to 0, thus push a new pointer to this vector, invalidating any iterators.

Reply:
Can't you just move the vector into a local, and iterate that?

Reply:
Can also ping pong two vectors with swap to avoid the reallocs. The nested loop should not be a problem should it?

Reply:
What would the code look like to be per-thread?  I know @gezalore correctly warns against premature optimization but this is an area I can guarantee we will have lock contention.

Reply:
It's not an issue to have an array of `VlDeleter`s, but then whenever an object refcount reaches 0, we'd have to put it in the current thread's deleter. However, that happens in the `VlClassRef` destructor (or assignment operator). So we'd need to use something like `VL_THREAD_ID() % threadCount` to know where to put the garbage object.